### PR TITLE
[DOC] update description and add behavior example for enable_cpu_pinning

### PR DESCRIPTION
**Pull request by wangleis**
### Details:
 - *update description and add behavior example for enable_cpu_pinning*

### Tickets:
 - *CVS-129650*

### CODE REVIEW

**Review by ilya-lavrenov — APPROVED** [review]

**Review by vurusovs — APPROVED** [review]

File: src/inference/include/openvino/runtime/properties.hpp [hunk 1]
```diff
@@ -403,8 +403,16 @@ static constexpr Property<SchedulingCoreType> scheduling_core_type{"SCHEDULING_C
  * @brief This property allows CPU threads pinning during inference.
  * @ingroup ov_runtime_cpp_prop_api
  *
- * Developer can use this property to use or not use CPU threads pinning during inference. If user does not explicitly
- * set value for this property, OpenVINO may choose any desired value based on internal logic.
+ * Developer can use this property to enable or disable CPU thread pinning during inference on Windows and Linux. MacOS
```
Comment (sunxiaoxia2022) [inline src/inference/include/openvino/runtime/properties.hpp:406]: Suggest changes `CPU thread pinning` to `CPU threads pinning`.

File: src/inference/include/openvino/runtime/properties.hpp [hunk 2]
```diff
@@ -403,8 +403,16 @@ static constexpr Property<SchedulingCoreType> scheduling_core_type{"SCHEDULING_C
...
+ * does not support CPU thread pinning, and this property is always disabled. If user does not explicitly set value for
```
Comment (sunxiaoxia2022) [inline src/inference/include/openvino/runtime/properties.hpp:407]: Suggest changes `CPU thread pinning` to `CPU threads pinning`.

File: src/inference/include/openvino/runtime/properties.hpp (hunk 1 above)
Comment (wangleis) [reply src/inference/include/openvino/runtime/properties.hpp]: Update to CPU pinning as common definition

File: src/inference/include/openvino/runtime/properties.hpp (hunk 2 above)
Comment (wangleis) [reply src/inference/include/openvino/runtime/properties.hpp]: Update to CPU pinning as common definition

**Review by xufang-lisa — APPROVED** [review]

**Comment by wangleis** [comment]
@dmitry-gorokhov Please take a look

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**